delta/histogram.go

```diff
@@ -102,23 +102,17 @@ func toHistogramKey(hist *collectors.HistogramMetric) uint64 {
 }
 
 func mergeHistograms(existing *collectors.HistogramMetric, current *collectors.HistogramMetric) *collectors.HistogramMetric {
```

Comment:
nit: This function is not optimal because `current` is passed as a pointer, modified and returned. You technically don't need to return it because you modify its values via the pointer. Although in its current usage it's ok, it can be harmful in the future because when you see `entry.Collected[key] = mergeHistograms(existing, currentValue)` you don't think that currentValue is modified. (this was raised during the meeting)
A better design IMO would be to change it to `func (h *HistogramMetric) MergeHistogram(other *HistogramMetric)`. Then you can do `existing.MergeHistogram(currentValue)` (I find it better to merge the currentValue to the existing rather than the existing in the currentValue, but both yield the same result)

Reply:
I do like `func (h *HistogramMetric) MergeHistogram(other *HistogramMetric)` much better and will do that refactor. I wrote it the way it is originally because I tend to dislike a function which mutates values and does not return the mutated value.

As far as the merge order I chose `existing` -> `current` because of a few of the extra fields on HistogramMetric like,
```go
	ReportTime     time.Time
	CollectionTime time.Time
``` 
where the value on `current` is what we need. It felt less accident prone in the future if a new field is added and isn't properly copied from `current` -> `existing` but could easily go either way on it 🤷 .

Reply:
makes sense, I missed that, thanks!